### DISCUSSION TITLE
Allow different output templates for writing tips to the console

### DISCUSSION THREAD

**Discussion by deadlydog** [deadlydog/PowerShell.tiPS#36] [Ideas]
Some users may want to see all of the tip info, others may only want to see the main info. Some users may want pretty headers and formatting and colours, others may want compact output. By providing a few templates for users to choose from, and potentially letting them define their own, they can customize the output to see the tips formatted how they like.

I'm curious how many people want this feature, so please upvote this if you do.

In addition, comment here on what you would like the end-user experience to look like. Things like:
- Just have a static list of 4 or 5 templates to choose from
- Allow users to create their own templates?

What should the template look like?
  - An XML or JSON file with the various properties and their values?
  - Properties stored in the tiPS Configuration file that they can set individually via a cmdlet?
  - A text file with tokens that define where each property is written to, and the color is should be written as?
  - Something else?

What should the template allow to be configured?
  - Just the text colours?
  - Which properties are written?
  - The formatting of the properties?
  - Additional things, like headers and their styles, whitespace, etc.

**Comment by peetrike** [comment]
You could do the formatting using PowerShell format files.  That allows people to come up and load their custom format files to change the result on screen.  Format file can also have several formats prebuilt.

You can look at the way how custom formatting is used to format help MAML to screen.  Or how [EZOut](https://github.com/StartAutomating/EZOut) module allows to create formatting
